src/SharpGLTF.Core/Schema2/agi.Articulations.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Numerics;
+using System.Linq;
+using SharpGLTF.Collections;
+
+namespace SharpGLTF.Schema2
+{
+    public sealed partial class AgiRootArticulations
+    {
+        internal AgiRootArticulations(ModelRoot root)
+        {
+            _articulations = new ChildrenCollection<AgiArticulation, AgiRootArticulations>(this);
+        }
+
+        protected override IEnumerable<ExtraProperties> GetLogicalChildren()
+        {
+            return base.GetLogicalChildren().Concat(_articulations);
+        }
+
+        public IReadOnlyList<AgiArticulation> Articulations => _articulations;
+
+        public AgiArticulation CreateArticulation(string name)
+        {
+            var articulation = new AgiArticulation();
+            articulation.Name = name;
+
+            _articulations.Add(articulation);
+
+            return articulation;
+        }
+    }
+
+    public sealed partial class AgiNodeArticulations
+    {
+        internal AgiNodeArticulations(Node node) { }
+
+        public String ArticulationName
+        {
+            get => _articulationName;
+            set => _articulationName = value;
+        }
+
+        public Boolean? IsAttachPoint
```

Comment:
I try to avoid nullable values because in the schema they imply "default value", but the user doesn't need to know which is that "default" value. From my point of view, it's an implementation detail imposed by the schema.

So along the library I use this pattern:

```c#

private const Boolean _IsAttachPointDefVal = false;

public Boolean IsAttachPoint
{
    get => _isAttachPoint.AsValue(_IsAttachPointDefVal );
    set => _isAttachPoint = value.AsNullable(_IsAttachPointDefVal);
}

Reply:
Done